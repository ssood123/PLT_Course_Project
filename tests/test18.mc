function void main()
{
  int array[6] arr;
  float array[7] arr1;
  int i;
  arr = (1,2,3,4,5,6);
  for (i = 0; i < lenArr(arr); i = i + 1) {
    print(arr[i]);
  }

  arr1 = (1.1,2.2,3.3,4.4,5.5,6.6,7.7);
  for (i = 0; i < lenArr(arr1); i = i + 1) {
    printf(arr1[i]);
  }
}
