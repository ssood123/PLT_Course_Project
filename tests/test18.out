; ModuleID = 'integraph'
source_filename = "integraph"

@fmt = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.1 = private unnamed_addr constant [4 x i8] c"%s\0A\00", align 1
@fmt.2 = private unnamed_addr constant [4 x i8] c"%g\0A\00", align 1

declare i32 @printf(i8*, ...)

declare i32 @printbig(i32)

define void @main() {
entry:
  %arr = alloca [6 x i32], align 4
  %arr1 = alloca [7 x double], align 8
  %i = alloca i32, align 4
  store [6 x i32] [i32 1, i32 2, i32 3, i32 4, i32 5, i32 6], [6 x i32]* %arr, align 4
  store i32 0, i32* %i, align 4
  br label %while

while:                                            ; preds = %while_body, %entry
  %i5 = load i32, i32* %i, align 4
  %tmp6 = icmp slt i32 %i5, 6
  br i1 %tmp6, label %while_body, label %merge

while_body:                                       ; preds = %while
  %i1 = load i32, i32* %i, align 4
  %arr2 = getelementptr [6 x i32], [6 x i32]* %arr, i32 0, i32 %i1
  %arr3 = load i32, i32* %arr2, align 4
  %printf = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt, i32 0, i32 0), i32 %arr3)
  %i4 = load i32, i32* %i, align 4
  %tmp = add i32 %i4, 1
  store i32 %tmp, i32* %i, align 4
  br label %while

merge:                                            ; preds = %while
  store [7 x double] [double 1.100000e+00, double 2.200000e+00, double 3.300000e+00, double 4.400000e+00, double 5.500000e+00, double 6.600000e+00, double 7.700000e+00], [7 x double]* %arr1, align 8
  store i32 0, i32* %i, align 4
  br label %while7

while7:                                           ; preds = %while_body8, %merge
  %i15 = load i32, i32* %i, align 4
  %tmp16 = icmp slt i32 %i15, 7
  br i1 %tmp16, label %while_body8, label %merge17

while_body8:                                      ; preds = %while7
  %i9 = load i32, i32* %i, align 4
  %arr110 = getelementptr [7 x double], [7 x double]* %arr1, i32 0, i32 %i9
  %arr111 = load double, double* %arr110, align 8
  %printf12 = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt.2, i32 0, i32 0), double %arr111)
  %i13 = load i32, i32* %i, align 4
  %tmp14 = add i32 %i13, 1
  store i32 %tmp14, i32* %i, align 4
  br label %while7

merge17:                                          ; preds = %while7
  ret void
}
